python_modules/dagster/dagster/_core/definitions/job_definition.py

```diff
@@ -832,6 +836,19 @@ def get_direct_input_values_from_job(target: PipelineDefinition) -> Mapping[str,
     description="Built-in filesystem IO manager that stores and retrieves values using pickling."
 )
 def default_job_io_manager(init_context: "InitResourceContext"):
+    # support overriding the default io manager via environment variables
+    module_name = os.getenv("DAGSTER_DEFAULT_IO_MANAGER_MODULE")
+    class_name = os.getenv("DAGSTER_DEFAULT_IO_MANAGER_CLASS")
+    if module_name and class_name:
+        from dagster._core.storage.io_manager import IOManager
+
+        io_manager_override = ConfigurableClassData(
+            module_name=module_name, class_name=class_name, config_yaml=yaml.dump({})
+        ).rehydrate()
+        check.invariant(isinstance(io_manager_override, IOManager))
+        return io_manager_override
```

Comment:
Can we use a single env var - since you need both for this to work?

Also can you add a note re why we need the `ConfigurableClassData` and it is not just a simple `import()` like the default one.

Reply:
> Can we use a single env var
Not sure I follow. Like concatenate the strings?

Reply:
It can get a little confusing because `.` is also the delimiter for components within a Python package path